src/windows/network_status.c

```diff
@@ -173,14 +187,45 @@ static void window_network_status_invalidate(rct_window *w)
 static void window_network_status_paint(rct_window *w, rct_drawpixelinfo *dpi)
 {
 	window_draw_widgets(w, dpi);
-	gCurrentFontSpriteBase = FONT_SPRITE_BASE_MEDIUM;
-	char buffer[sizeof(window_network_status_text) + 10];
-	char* lineCh = buffer;
-	lineCh = utf8_write_codepoint(lineCh, FORMAT_BLACK);
-	strcpy(lineCh, window_network_status_text);
-	gfx_clip_string(buffer, w->widgets[WIDX_BACKGROUND].right - 50);
+	
 	int x = w->x + (w->width / 2);
 	int y = w->y + (w->height / 2);
-	x -= gfx_get_string_width(buffer) / 2;
-	gfx_draw_string(dpi, buffer, 0, x, y);
+	int width = w->width - (24 + 13);
+	int colour = COLOUR_BLACK;
+
+	//Below is a modified version of gfx_draw_string_centred_wrapped
+	int font_height, line_height, line_width, line_y, num_lines;
+	char* buffer = RCT2_ADDRESS(0x009C383D, char);
+
+	if (gCurrentFontSpriteBase >= 0) {
+		gCurrentFontSpriteBase = FONT_SPRITE_BASE_MEDIUM;
+	}
+
+	gfx_draw_string(dpi, buffer, colour, dpi->x, dpi->y);
+
+	buffer = RCT2_ADDRESS(RCT2_ADDRESS_COMMON_STRING_FORMAT_BUFFER, char);
+	format_string_raw(buffer, window_network_status_text, NULL);
+
+	gCurrentFontSpriteBase = FONT_SPRITE_BASE_MEDIUM;
+
+	// line_width unused here
+	line_width = gfx_wrap_string(buffer, width, &num_lines, &font_height);
+	line_height = LINE_HEIGHT;
+
+	if (*buffer == FORMAT_OUTLINE) {
+		line_height = line_height + 1;
+	}
+
+	font_height = (line_height / 2) * num_lines;
+	line_y = y - font_height;
+
+	gCurrentFontFlags = 0;
+
+	for (int line = 0; line <= num_lines; ++line) {
+		int half_width = gfx_get_string_width(buffer) / 2;
+		gfx_draw_string(dpi, buffer, 0xFE, x - half_width, line_y);
```

Comment:
Marijnvdwerf cleans up the String ID and strings right now, giving them a direct corresponding definition which can be found in 'string_ids'
I presume he means 0xFE should have a name, might be wrong :p